designs/SearchCommandSpec.md

```diff
@@ -0,0 +1,98 @@
+--------------------
+> Microsoft.Extensions.Logging.Abstractions | v5.0.0-preview.6.20305.6 | DLs: 345145935
+Logging abstractions for Microsoft.Extensions.Logging.
+
+Commonly Used Types:
+Microsoft.Extensions.Logging.ILogger
+Microsoft.Extensions.Logging.ILoggerFactory
+Microsoft.Extensions.Logging.ILogger&lt;TCategoryName&gt;
+Microsoft.Extensions.Logging.LogLevel
+Microsoft.Extensions.Logging.Logger&lt;T&gt;
+Microsoft.Extensions.Logging.LoggerMessage
+Microsoft.Extensions.Logging.Abstractions.NullLogger
+
+When using NuGet 3.x this package requires at least version 3.4.
+
+--------------------
+> Microsoft.Extensions.Logging | v5.0.0-preview.6.20305.6 | DLs: 243186566
+Logging infrastructure default implementation for Microsoft.Extensions.Logging.
+When using NuGet 3.x this package requires at least version 3.4.
+
+--------------------
+> Microsoft.IdentityModel.Logging | v6.6.0 | DLs: 120177247
+Includes Event Source based logging support.
+
+--------------------
+> Microsoft.Extensions.Logging.Configuration | v5.0.0-preview.6.20305.6 | DLs: 63205816
+Configuration support for Microsoft.Extensions.Logging.
+When using NuGet 3.x this package requires at least version 3.4.
+
+--------------------
+> Microsoft.Extensions.Logging.Console | v5.0.0-preview.6.20305.6 | DLs: 61796431
+Console logger provider implementation for Microsoft.Extensions.Logging.
+When using NuGet 3.x this package requires at least version 3.4.
+
+--------------------
+> Microsoft.Extensions.Logging.Debug | v5.0.0-preview.6.20305.6 | DLs: 58643995
+Debug output logger provider implementation for Microsoft.Extensions.Logging. This logger logs messages to a debugger monitor by writing messages with System.Diagnostics.Debug.WriteLine().
+When using NuGet 3.x this package requires at least version 3.4.
+
+--------------------
+> Microsoft.Extensions.Logging.EventSource | v5.0.0-preview.6.20305.6 | DLs: 32121683
+EventSource/EventListener logger provider implementation for Microsoft.Extensions.Logging.
+When using NuGet 3.x this package requires at least version 3.4.
+
+--------------------
+> Serilog.Extensions.Logging | v3.0.2-dev-10280 | DLs: 43115462
+-- More  --
+```
+
+## Future Work
+
+* Enabling cross-platform functionality for the search feature
+* Adding optional arguments that allow users to customize their search queries
+
+## Open Questions
+
+* What represents the most effective design choice for displaying the search results, in terms of spacing, demarcation, truncating package descriptions, and so on?
+
+## Considerations
+
+### 1. Combining the results from different sources
+
+Initially, we considered consolidating all search results from the various sources and displaying them together, rather than separating them based on their source. However, users may wish to know where a particular package is located. Also, combining the search results for different sources leads to the complex problem of how to combine these results most effectively.
```

Comment:
I definitely agree with this section. One thing I'll call out is that NuGet normally uses your `nuget.config` file to discover package sources to read from. This search experience won't do that, which may confuse users. Is this something we want to address somehow?

Reply:
Where will it get its sources, then?

Reply:
I took this to mean the sources still come from config, but they are not interleaved.

If there is one "section" per source, how will they be sorted? lex by name? lex by URL? same order as config?

Reply:
This section was meant to convey that instead of interleaving the results from the various sources in nuget.config, we display the results for each source as a separate section, followed by the results for the next source, and so on. As far as I'm aware, the NuGet API's Search only returns results sorted by relevance. I have also added a -Source optional argument in the 'Optional Arguments' section in 'Solution' if a user wants to specify a source for the Search to target.